test/src/code_field/tabs_to_spaces_test.dart

```diff
@@ -0,0 +1,223 @@
+      const cursorPosition = 5;
+      TextEditingValue value = const TextEditingValue(
+        text: _codeRowWithoutTabs,
+        selection: TextSelection.collapsed(offset: cursorPosition),
+      );
+
+      value = value.tabsToSpaces(_spaceCount);
+
+      expect(value.text, _codeRowWithoutTabs);
+      expect(value.selection.baseOffset, cursorPosition);
+      expect(value.selection.extentOffset, cursorPosition);
+    });
+
+    test('Code row without tabs with cursor at end', () {
+      const cursorPosition = _codeRowWithoutTabs.length;
+      TextEditingValue value = const TextEditingValue(
+        text: _codeRowWithoutTabs,
+        selection: TextSelection.collapsed(offset: cursorPosition),
+      );
+
+      value = value.tabsToSpaces(_spaceCount);
+
+      expect(value.text, _codeRowWithoutTabs);
+      expect(value.selection.baseOffset, cursorPosition);
+      expect(value.selection.extentOffset, cursorPosition);
+    });
+
+    test('Code row without tabs with selection', () {
+      const selectionStart = 5;
+      const selectionEnd = 10;
+      TextEditingValue value = const TextEditingValue(
+        text: _codeRowWithoutTabs,
+        selection: TextSelection(
+          baseOffset: selectionStart,
+          extentOffset: selectionEnd,
+        ),
+      );
+
+      value = value.tabsToSpaces(_spaceCount);
+
+      expect(value.text, _codeRowWithoutTabs);
+      expect(value.selection.baseOffset, selectionStart);
+      expect(value.selection.extentOffset, selectionEnd);
+    });
+
+    test('Code row without tabs with reversed selection', () {
+      const selectionStart = 10;
+      const selectionEnd = 5;
+      TextEditingValue value = const TextEditingValue(
+        text: _codeRowWithoutTabs,
+        selection: TextSelection(
+          baseOffset: selectionStart,
+          extentOffset: selectionEnd,
+        ),
+      );
+
+      value = value.tabsToSpaces(_spaceCount);
+
+      expect(value.text, _codeRowWithoutTabs);
+      expect(value.selection.baseOffset, selectionStart);
+      expect(value.selection.extentOffset, selectionEnd);
+    });
+
+    test('Code with tabs without selection', () {
+      TextEditingValue value = const TextEditingValue(text: _codeWithTabs);
+      
+      value = value.tabsToSpaces(_spaceCount);
+      
+      expect(value.text, _codeWithDoubleSpaces);
+      expect(value.selection.baseOffset, -1);
+      expect(value.selection.extentOffset, -1);
+    });
+
+    test('Code with tabs with cursor at start', () {
+      TextEditingValue value = const TextEditingValue(
+        text: _codeWithTabs,
+        selection: TextSelection.collapsed(offset: 0),
+      );
+      
+      value = value.tabsToSpaces(_spaceCount);
+      
+      expect(value.text, _codeWithDoubleSpaces);
+      expect(value.selection.baseOffset, 0);
+      expect(value.selection.extentOffset, 0);
+    });
+
+    test('Code with tabs with cursor at middle', () {
+      const cursorPosition = _codeWithTabs.length ~/ 2;
```

Comment:
Hardcode a number please so it can be easily verified.
Then there's no need for a comment below.